internal/spiffe/spiffe.go

```diff
@@ -0,0 +1,77 @@
+package spiffe
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+)
+
+type SVIDDetails struct {
+	PrivateKey    crypto.Signer
+	Certificate   *x509.Certificate
+	Intermediates []*x509.Certificate
+}
+
+type SVIDFetcher interface {
```

Comment:
Would be cool to have some docstrings here. What do you think? :)

Reply:
Added missing docstring.